python/ray/dashboard/subprocesses/module.py

```diff
@@ -47,12 +52,38 @@ class SubprocessModule(abc.ABC):
     def __init__(
         self,
         config: SubprocessModuleConfig,
+        parent_process_pid: int,
     ):
         """
         Initialize current module when DashboardHead loading modules.
         :param dashboard_head: The DashboardHead instance.
         """
         self._config = config
+        self._parent_process = psutil.Process(parent_process_pid)
```

Comment:
What if parent is already dead and the pid is reused?

Reply:
This is in the constructor of the child process. So do you mean that the parent process dies immediately even before the child process constructor has run?

Reply:
I've changed the implementation to use `multiprocessing.parent_process().is_alive()` because I found that the `parent_process()` call returns a variable maintained by Python, so it should be able to identify the parent process calling `multiprocessing`.  

https://github.com/python/cpython/blob/a936af924efc6e2fb59e27990dcd905b7819470a/Lib/multiprocessing/process.py#L51-L55